File: python/hackerrank/simulation/viralAdvertising.py

def viralAdvertising(n):
    """
    Calculates the cumulative number of likes for a viral advertising campaign.
    
    The campaign starts by sharing with 5 people. Each day, floor(shared/2) people 
    like the advertisement and share it with 3 friends each the following day.
    
    Args:
        n (int): The number of days to track.
        
    Returns:
        int: The total (cumulative) number of likes after n days.
    """
    shared = 5
    cumulative_likes = 0
    
    for _ in range(n):
        liked = shared // 2
        cumulative_likes += liked
        shared = liked * 3
        
    return cumulative_likes

def main():
    """
    Reads input from stdin and prints the result of viralAdvertising.
    """
    try:
        n = int(input().strip())
        print(viralAdvertising(n))
    except EOFError:
        pass

if __name__ == '__main__':
    main()

File: golang/hackerrank/simulation/viralAdvertising.go
package simulation

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// ViralAdvertising calculates cumulative likes for a campaign.
func ViralAdvertising(n int) int {
	shared := 5
	cumulativeLikes := 0

	for i := 0; i < n; i++ {
		liked := shared / 2
		cumulativeLikes += liked
		shared = liked * 3
	}

	return cumulativeLikes
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	line, _ := reader.ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(line))
	fmt.Println(ViralAdvertising(n))
}
